packages/discord.js/src/managers/GuildManager.js

```diff
@@ -84,7 +84,7 @@ class GuildManager extends CachedManager {
    * @property {Snowflake|number} [parentId] The parent id for this channel
    * @property {ChannelType|number} [type] The type of the channel
    * @property {string} name The name of the channel
-   * @property {string} [topic] The topic of the text channel
+   * @property {?string} [topic] The topic of the text channel
```

Comment:
Why is it `?string` here but `string|null` in ChannelData

Reply:
Yeah you're right, the partial data can be `undefined | string | null` (`null` being no topic set). Is there a definition for Partial Channel Data? The developer portal doesn't say anything and it is inconsistently defined inside discord.js. The typedef says that topic is always string while TS typing says it can be undefined.
https://github.com/discordjs/discord.js/blob/68d5169f66c96f8fe5be17a1c01cdd5155607ab2/packages/discord.js/src/managers/GuildManager.js#L79-L96
https://github.com/discordjs/discord.js/blob/68d5169f66c96f8fe5be17a1c01cdd5155607ab2/packages/discord.js/typings/index.d.ts#L4825-L4847

Reply:
> The typedef says that topic is always string while TS typing says it can be undefined.

Incorrect - if you look closely, the topic in the JSDoc is wrapped in `[this]` which indicates an optional parameter, which is reflected accordingly in the typings. Not sure about needing to pass `null`, though... why pass `null` to create a channel alongside a topic when one can omit it?

Reply:
> Here's a better example, a non-partial non-category channel's parent value is a CategoryChannel or it's null. If we wanted to save a channel as JSON and use it to create new channels we face the issue that parent accepts CategoryChannelResolvable | undefined. Of which our channel parent value might be neither because it could be null. Sure you could just transform it to undefined but to me it doesn't really make sense to not allow null (because it works and it's accepted by Discord API). It just seems inconsistent.
Same as this. Omitting it and setting it to null does the same thing during creation and that's why I find it sensible to allow both rather then omitting it every time that I'm making a channel with no topic.